src/node_zlib.cc

```diff
@@ -254,11 +253,22 @@ class ZCtx : public AsyncWrap {
             ctx->err_ = Z_NEED_DICT;
           }
         }
+        if (ctx->strm_.avail_in >= 2 && ctx->mode_ == GUNZIP) {
```

Comment:
Yeah, I thought this was a little silly. It should probably check to make sure that `avail_in` is at least equal to the gzip header size. I'll fix this.